source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,871 @@
+cursor. Drivers MUST NOT append a ``maxTimeMS`` field to ``getMore``
+commands.
+
+See `Non-tailable cursor behavior`_.
+
+Tailable Cursors
+~~~~~~~~~~~~~~~~
+
+Tailable cursors only support the ``ITERATION`` value for the ``timeoutMode``
+option. This is the default value and drivers MUST error if the option is set
+to ``CURSOR_LIFETIME``.
+
+Tailable non-awaitData Cursors
+``````````````````````````````
+
+If ``timeoutMS`` is set, drivers MUST apply it separately to the original
+operation and to all ``next`` calls on the resulting cursor but MUST NOT
+append a ``maxTimeMS`` field to any commands.
+
+Tailable awaitData Cursors
+``````````````````````````
+
+If ``timeoutMS`` is set, drivers MUST apply it to the original operation.
+Drivers MUST also apply the original ``timeoutMS`` value to each ``next``
+call on the resulting cursor but MUST NOT use it to derive a ``maxTimeMS``
+value for ``getMore`` commands. Helpers for operations that create tailable
+awaitData cursors MUST also support the ``maxAwaitTimeMS`` option. Drivers
+MUST error if this option is set, ``timeoutMS`` is set to a non-zero value,
+and ``maxAwaitTimeMS`` is greater than or equal to ``timeoutMS``. If this
+option is set, drivers MUST use it as the ``maxTimeMS`` field on ``getMore``
+commands.
+
+See `Tailable cursor behavior`_ for rationale regarding both non-awaitData
+and awaitData cursors.
+
+Change Streams
+~~~~~~~~~~~~~~
+
+Driver ``watch`` helpers MUST support both ``timeoutMS`` and
+``maxAwaitTimeMS`` options. Drivers MUST error if ``maxAwaitTimeMS`` is set,
+``timeoutMS`` is set to a non-zero value, and ``maxAwaitTimeMS`` is greater
+than or equal to ``timeoutMS``. If set, drivers MUST apply the ``timeoutMS``
+option to the initial ``aggregate`` operation. Drivers MUST also apply the
+original ``timeoutMS`` value to each ``next`` call on the change stream but
+MUST NOT use it to derive a ``maxTimeMS`` field for ``getMore`` commands. If
+the ``maxAwaitTimeMS`` option is set, drivers MUST use it as the
+``maxTimeMS`` field on ``getMore`` commands.
+
+If a ``next`` call fails with a timeout error, drivers MUST NOT invalidate
+the change stream. The subsequent ``next`` call MUST perform a resume attempt
+to establish a new change stream on the server. Any errors from the
+``aggregate`` operation done to create a new change stream MUST be propagated
+to the application. Drivers MUST document that users can either call ``next``
+again or close the existing change stream and create a new one if a previous
+``next`` call times out. The documentation MUST suggest closing and
+re-creating the stream with a higher timeout if the timeout occurs before any
+events have been received because this is a signal that the server is timing
+out before it can finish processing the existing oplog.
+
+See `Change stream behavior`_.
+
+Convenient Transactions API
+---------------------------
+
+If ``timeoutMS`` is set, drivers MUST apply it to the entire
```

Comment:
Can this be set anywhere but globally?  I don't recall.

Reply:
It can only be set globally. We allow a timeout to be set on the session, but that's an internal mechanism for `withTransaction` to communicate timeouts to operations in the user-provided callback.

Reply:
I thought you said elsewhere that it would be an additional parameter to `withTransaction`.

Reply:
Ah, right. It is inherited from the MongoClient but can also be overridden for the operation via the new `WithTransactionOptions` type.